src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -0,0 +1,152 @@
+import slimeknights.tconstruct.library.modifiers.modules.util.ModuleBuilder;
+import slimeknights.tconstruct.library.module.HookProvider;
+import slimeknights.tconstruct.library.module.ModuleHook;
+import slimeknights.tconstruct.library.module.ModuleHookMap;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataCapability;
+import slimeknights.tconstruct.library.tools.context.EquipmentChangeContext;
+import slimeknights.tconstruct.library.tools.nbt.IToolStackView;
+import slimeknights.tconstruct.library.utils.IdParser;
+import slimeknights.tconstruct.library.utils.Util;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public record MaxArmorStatModule(TinkerDataCapability.TinkerDataKey<Float> statKey, LevelingValue amount, TinkerDataCapability.ComputableDataKey<ModifierMaxLevel> maxLevel, boolean allowBroken, @Nullable TagKey<Item> heldTag, ArmorStatModule.TooltipStyle tooltipStyle, ModifierCondition<IToolStackView> condition) implements HookProvider, EquipmentChangeModifierHook, ModifierModule, TooltipModifierHook, ModifierCondition.ConditionalModule<IToolStackView> {
+  private static final List<ModuleHook<?>> TOOLTIP_HOOKS = HookProvider.<ArmorStatModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE, ModifierHooks.TOOLTIP);
+  private static final List<ModuleHook<?>> NO_TOOLTIP_HOOKS = HookProvider.<ArmorStatModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE);
+  public static final RecordLoadable<MaxArmorStatModule> LOADER = RecordLoadable.create(
+    new IdParser<>(ResourceLocation::new, "Stat").xmap((location, factory) -> TinkerDataCapability.TinkerDataKey.<Float>of(location), (tinkerDataKey, errorFactory) -> tinkerDataKey.getId()).requiredField("maxLevel", MaxArmorStatModule::statKey),
+    LevelingValue.LOADABLE.directField(MaxArmorStatModule::amount),
+    BooleanLoadable.INSTANCE.defaultField("allow_broken", false, MaxArmorStatModule::allowBroken),
+    Loadables.ITEM_TAG.nullableField("held_tag", MaxArmorStatModule::heldTag),
+    new EnumLoadable<>(ArmorStatModule.TooltipStyle.class).defaultField("tooltip_style", ArmorStatModule.TooltipStyle.NONE, MaxArmorStatModule::tooltipStyle),
+    ModifierCondition.TOOL_FIELD,
+    MaxArmorStatModule::new);
+
+  public MaxArmorStatModule(TinkerDataCapability.TinkerDataKey<Float> statKey, LevelingValue amount, boolean allowBroken, @Nullable TagKey<Item> heldTag, ArmorStatModule.TooltipStyle tooltipStyle, ModifierCondition<IToolStackView> condition) {
+    this(statKey, amount, TinkerDataCapability.ComputableDataKey.of(IdExtender.LocationExtender.INSTANCE.prefix(statKey.getId(), "_data"), ModifierMaxLevel::new), allowBroken, heldTag, tooltipStyle, condition);
+  }
+
+  @Override
+  public RecordLoadable<MaxArmorStatModule> getLoader() {
+    return LOADER;
+  }
+
+  @Override
+  public void addModules(ModuleHookMap.Builder builder) {
+    builder.addModule(new MaxArmorLevelModule(maxLevel, allowBroken, heldTag));
+  }
```

Comment:
Since you update the max level in the equipment change hooks, this module is not actually needed, so you can ditch this call. Might mean `MaxArmorLevelModule` ended up unused, hold onto it for now though just in case it ends up being helpful for attributes.